doc/sphinx/language/core/basic.rst

```diff
@@ -0,0 +1,524 @@
+
+This section presents the most essential notions to understand the
+rest of the |Coq| manual: :term:`terms <term>` and :term:`types
+<type>` on the one hand, :term:`commands <command>` and :term:`tactics
+<tactic>` on the other hand.
+
+.. glossary::
+
+   term
+
+     Terms are the basic expressions of |Coq|.  Terms can represent
+     mathematical expressions, propositions and proofs, but also
+     executable programs and program types.
+
+     Here is the top-level syntax of terms.  Each of the listed
+     constructs is presented in a dedicated section.  Some of these
+     constructs (like :n:`@term_forall_or_fun`) are part of the core
+     language that the kernel of |Coq| understands and are therefore
+     described in :ref:`this chapter <core-language>`, while
+     others (like :n:`@term_if`) are language extensions that are
+     presented in :ref:`the next chapter <extensions>`.
+
+     .. insertprodn term qualid_annotated
+
+     .. prodn::
+        term ::= @term_forall_or_fun
+        | @term_let
+        | @term_if
+        | @term_fix
+        | @term_cofix
+        | @term100
+        term100 ::= @term_cast
+        | @term10
+        term10 ::= @term_application
+        | @one_term
+        one_term ::= @term_explicit
+        | @term1
+        term1 ::= @term_projection
+        | @term_scope
+        | @term0
+        term0 ::= @qualid_annotated
+        | @sort
+        | @primitive_notations
+        | @term_evar
+        | @term_match
+        | @term_record
+        | @term_generalizing
+        | @term_ltac
+        | ( @term )
+        qualid_annotated ::= @qualid {? @univ_annot }
+
+     .. note::
+
+        Many :term:`commands <command>` and :term:`tactics <tactic>`
+        use :n:`@one_term` (in the syntax of their arguments) rather
+        than :n:`@term`.  The former need to be enclosed in
+        parentheses unless they're very simple, such as a single
+        identifier.  This avoids confusing a space-separated list of
+        terms or identifiers with a :n:`@term_application`.
+
+   type
+
+     To be valid and accepted by the |Coq| kernel, a term needs an
+     associated type.  We express this relationship by “:math:`x` *of
+     type* :math:`T`”, which we write as “:math:`x:T`”.  Informally,
+     “:math:`x:T`” can be thought as “:math:`x` *belongs to*
+     :math:`T`”.
+
+     The |Coq| kernel is a type checker: it verifies that a term has
+     the expected type by applying a set of typing rules (see
+     :ref:`Typing-rules`).  If that's indeed the case, we say that the
+     term is :gdef:`well-typed`.
+
+     A special feature of the |Coq| language is that types can depend
+     on terms (we say that the language is `dependently-typed
+     <https://en.wikipedia.org/wiki/Dependent_type>`_).  Because of
+     this, types and terms share a common syntax.  All types are terms,
+     but not all terms are types:
+
+     .. insertprodn type type
+
+     .. prodn::
+        type ::= @term
```

Comment:
Is the objective here mainly to justify the `type` terminology. If yes, this can certainly be shortened.

Reply:
It is in lieu of a proper introduction for types.